test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,59 @@
+
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, key, values, concurrency):
+    logger.info(f"Starting writes for key {key} with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        while i < values:
+            await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, {key + i})")
+            i += concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes for key {key} with concurrency {concurrency}")
+
+async def wait_for_view(cql, base_table, view_table):
+    logger.info(f"Waiting for view {view_table} to be built")
+    while True:
+        base_count = await cql.run_async(f"SELECT count(*) FROM {base_table} USING TIMEOUT 1h", timeout=600)
+        view_count = await cql.run_async(f"SELECT count(*) FROM {view_table} USING TIMEOUT 1h", timeout=600)
+        if base_count[0][0] == view_count[0][0]:
+            logger.info(f"View {view_table} has been built")
+            return
+        time.sleep(0.2)
```

Comment:
Two comments on this function:
1. It is wasteful to scan the base table every time around this loop - the base-table count won't change.
2. The way you wait for the view update by reading the entire table is not the most efficient way you could have done. The most efficient thing you could have is to write the base table data *before* adding and only then add the view - at this point "view building" begins, and you can learn when view building finished by looking at a system table (tell me if you need a pointer). You don't need to read all the data to tell that.

That being said, if the base table is not huge, your method is acceptable too. It all depends if this test takes 2 seconds (when performance is not very important) or 10 minutes.

Reply:
I tried to use your suggestions for checking if views are built, please take a look